amethyst_rendy/src/lib.rs

```diff
@@ -1,6 +1,41 @@
-// this is temporary
-// #![allow(dead_code)]
-// #![allow(unused_variables)]
+//! This implementation of the Amethyst Renderer utilizes the `rendy` crate, built on top of
+//! `gfx-hal` to provide the building blocks for a AAA configurable rendering graph-based pipeline.
+//!
+//! As a general overview, this crate can be broken down as follows:
+//!
+//! ### Core
+//!
+//! ### Submodules
+//!
+//! ### Passes
+//! * [pass::flat2d::DrawFlat2DDesc]
+//! * [pass::flat2d::DrawFlat2DTransparentDesc]
+//! * [pass::pbr::DrawPbrDesc]
+//! * [pass::flat::DrawFlatDesc]
+//! * [pass::shaded::DrawShadedDesc]
+//! * [pass::skybox::DrawSkyboxDesc]
+//! * [pass::debug_lines::DrawDebugLinesDesc]
+//!
+//! ## Systems
+//! * [system::RenderingSystem]
+//! * [visibility::VisibilitySortingSystem]
+//! * [sprite_visibility::SpriteVisibilitySortingSystem]
+//!
+//! ## Components
+//! * [camera::Camera]
+//! * [sprite_visibility::SpriteVisibility]
+//! * [visibility::Visibility]
+//! * [visibility::BoundingSphere]
+//! * [debug_drawing::DebugLinesComponent]
+//! * [light::Light]
+//! * [resources::Tint]
+//! * [skinning::JointTransforms]
+//! * [sprite::SpriteRender]
+
+#![allow(dead_code)]
+#![allow(unused_variables)]
+#![deny(missing_docs)]
```

Comment:
Please lessen to `warn`, we are denying warnings on CI, but don't want to require fixing them when working locally.